aya/src/bpf.rs

```diff
@@ -580,12 +605,22 @@ impl<'a> BpfLoader<'a> {
                 (name, program)
             })
             .collect();
-        let maps: Result<HashMap<String, Map>, BpfError> = maps.drain().map(parse_map).collect();
+        let maps = maps
+            .drain()
+            .map(parse_map)
+            .collect::<Result<HashMap<String, Map>, BpfError>>()?;
+
+        if !self.allow_unsupported_maps
+            && maps
+                .iter()
+                .filter(|(_, x)| matches!(x, Map::Unsupported(_)))
+                .count()
+                != 0
+        {
+            return Err(BpfError::UnsupportedMap);
```

Comment:
This should be BpfError::MapError(MapError::Unsupported)

Reply:
Done